CRM/Contact/BAO/Query.php

```diff
@@ -4387,11 +4387,14 @@ protected static function convertCustomDateRelativeFields(&$formValues, &$params
         return;
       }
 
-      list($from, $to) = CRM_Utils_Date::getFromTo(
-        NULL,
-        (empty($formValues[$customFieldName . '_from']) ? NULL : $formValues[$customFieldName . '_from']),
-        CRM_Utils_Array::value($customFieldName . '_to', $formValues)
-      );
+      $from = CRM_Utils_Array::value($customFieldName . '_from', $formValues, NULL);
+      $to = CRM_Utils_Array::value($customFieldName . '_to', $formValues, NULL);
+
+      $customFieldID = CRM_Core_BAO_CustomField::getKeyID($customFieldName);
+      $customFieldType = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_CustomField', $customFieldName, 'data_type');
+      if ($customFieldType == 'Date') {
+        list($from, $to) = CRM_Utils_Date::getFromTo(NULL, $from, $to);
+      }
     }
```

Comment:
You added some unit tests to this in the past didn't you? Doesn't look like this patch will be backportable to 4.6 which is also affected :-(